libr/io/io_map.c

```diff
@@ -594,8 +594,15 @@ R_API bool r_io_map_write_to_overlay(RIOMap *map, ut64 addr, const ut8 *buf, int
 	}
 	if (r_itv_begin (chunk->itv) < r_itv_begin (search_itv)) {
 		chunk->itv.size = r_itv_begin (search_itv) - r_itv_begin (chunk->itv);
-		// realloc cannot fail here because the new size is smaller than the old size
-		chunk->buf = realloc (chunk->buf, r_itv_size (chunk->itv) * sizeof (ut8));
+		st64 new_size = r_itv_size (chunk->itv) * sizeof (ut8);
+		if (new_size > 0) {
+			ut8 *chunk_buf = realloc (chunk->buf, new_size);
+			if (chunk_buf) {
```

Comment:
cannot fail, because new size is always smaller than old size. realloc cannot run out of memory here

Reply:
thats up to the heap implementation. better dont assume what the heap allocator does before hand. not all builds of r2 use glibc

Reply:
uhm, ok